include/ozo/detail/begin_statement_builder.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <ozo/core/none.h>
+#include <ozo/core/options.h>
+#include <ozo/query_builder.h>
+#include <ozo/transaction_options.h>
+#include <type_traits>
+
+namespace ozo::detail {
+
+struct begin_statement_builder {
+    constexpr static auto empty_sql() {
+        using namespace ozo::literals;
+        return ""_SQL;
+    }
+
+    template <typename Level, typename = std::enable_if_t<is_isolation_level_v<Level>>>
+    constexpr static auto level_part(Level) {
+        using namespace ozo::literals;
+
+        auto prefix = " ISOLATION LEVEL "_SQL;
+        if constexpr (std::is_same_v<Level, isolation_level::serializable_t>) {
+            return std::move(prefix) + "SERIALIZABLE"_SQL;
+        } else if constexpr (std::is_same_v<Level, isolation_level::repeatable_read_t>) {
+            return std::move(prefix) + "REPEATABLE READ"_SQL;
+        } else if constexpr (std::is_same_v<Level, isolation_level::read_committed_t>) {
+            return std::move(prefix) + "READ COMMITTED"_SQL;
+        } else {
+            static_assert(std::is_same_v<Level, isolation_level::read_uncommitted_t>, "unexpected isolation level");
+            return std::move(prefix) + "READ UNCOMMITTED"_SQL;
+        }
+    }
+
+    constexpr static auto level_part(none_t) {
+        return empty_sql();
+    }
+
+    template <typename Mode, typename =  std::enable_if_t<is_transaction_mode_v<Mode>>>
+    constexpr static auto mode_part(Mode) {
+        using namespace ozo::literals;
+
+        if constexpr (std::is_same_v<Mode, transaction_mode::read_write_t>) {
+            return " READ WRITE"_SQL;
+        } else {
+            static_assert(std::is_same_v<Mode, transaction_mode::read_only_t>, "unexpected transaction mode");
+            return " READ ONLY"_SQL;
+        }
+    }
+
+    constexpr static auto mode_part(none_t) {
+        return empty_sql();
+    }
+
+    template <typename Deferrable, typename = std::enable_if_t<is_deferrable_mode_v<Deferrable>>>
+    constexpr static auto deferrable_part(Deferrable deferrable) {
+        using namespace ozo::literals;
+
+        auto deferrable_sql = " DEFERRABLE"_SQL;
+        if constexpr (!deferrable()) {
+            return " NOT"_SQL + std::move(deferrable_sql);
+        } else {
+            return deferrable_sql;
+        }
```

Comment:
In the same way there maybe something like this:

```c++
template <typename Deferrable>
constexpr static auto deferrable_part(Deferrable deferrable) {
    if constexpr (decltype(level == none)::value) {
        return empty_sql;
    } else if constexpr (Deferrable::value) {
       return deferrable_sql;
    } else {
        return BOOST_HANA_STRING(" NOT") + deferrable_sql;
    }
}
```

Reply:
I did it in this way, but this actually fails for the case when level is indeed none, as the comparison operator for `none_t` is called, thus resulting in bool being decltyped.

I fixed this and the other stuff above accordingly, but for the none-case I simply sticked to calling `is_none<T>::value` for `Deferrable`, `Mode` and `Level`.